charts/k8ssandra-cluster/templates/stargate/stargate.yaml

```diff
@@ -0,0 +1,109 @@
+{{- if .Values.stargate.enabled }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ .Release.Name }}-k8ssandra-stargate
+  labels:
+    app: {{ .Release.Name }}-k8ssandra-stargate
+{{ include "k8ssandra-cluster.labels" . | indent 4 }}
+spec:
+  replicas: {{ default 1 .Values.stargate.replicas }}
+  selector:
+    matchLabels:
+      app: {{ .Release.Name }}-k8ssandra-stargate
+  template:
+    metadata:
+      labels:
+        app: {{ .Release.Name }}-k8ssandra-stargate
+    spec:
+      initContainers:
+          - name: wait-for-cassandra
+            image: alpine:3.12.2
+            imagePullPolicy: IfNotPresent
+            command:
+              - sh
+            args:
+              - -c
+              - |
+                echo "Waiting for at least one cassandra node to finish bootstrapping..."
+                while ! nslookup {{ .Values.clusterName }}-seed-service.{{ .Release.Namespace }}.svc.{{ .Values.clusterDomain | default "cluster.local" }}; do
+                    sleep 5
+                done
+                echo "Cassandra is ready. Starting Stargate..."
+      containers:
+        - name: {{ .Release.Name }}-k8ssandra-stargate
+  {{- if eq .Values.stargate.containerImage "auto" }}
+    {{- if eq .Values.stargate.clusterVersion "3.11" }}
+          image: "stargateio/stargate-3_11:v1.0.0"
+    {{- else if eq .Values.stargate.clusterVersion "4.0" }}
+          image: "stargateio/stargate-4_0:v1.0.0"
+    {{- else }}
+      {{- fail tpl "clusterVersion {{ .Values.stargate.clusterVersion }} is not supported." }}
```

Comment:
When I use a bad `clusterVersion`, `fail` fails :)

```
$ helm template bad-stargate ./k8ssandra-cluster/ -f stargate-values.yaml
Error: template: k8ssandra-cluster/templates/stargate/stargate.yaml:41:10: executing "k8ssandra-cluster/templates/stargate/stargate.yaml" at <fail>: wrong number of args for fail: want 1 got 2
```
It looks like the right syntax is,

```
{{- fail "clusterVersion .Values.stargate.clusterVersion is not supported." }}
```

Reply:
ah, thanks. I forgot to test that. 😬